src/Interpreters/Aggregator.cpp

```diff
@@ -1067,13 +1094,13 @@ void NO_INLINE Aggregator::executeImpl(
 
     if (use_cache)
     {
-        typename Method::State state(key_columns, key_sizes, aggregation_state_cache);
+        typename Method::State state(key_columns, key_sizes, aggregation_state_cache, reinterpret_cast<UInt64>(&result));
```

Comment:
I'm a little bit confused by this code. IMO it would be natural if all the new logic will be contained within the new aggregation method and fallback from low-cardinality to a general case will be implemented somehow similar in principle to single-level -> two-level transition.
and the fact that we use `AggregatedDataVariants`'s address here to find a relevant `AggregatorThreadContext` is an evidence that it prob should be just a member of `AggregatedDataVariants`. then I guess code will look simpler.

Reply:
it's a good idea

Reply:
add a new member `methond_context` in  `AggregatedDataVariants `